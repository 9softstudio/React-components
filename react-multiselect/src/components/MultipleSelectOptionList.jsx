import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import Option from './Option'

export default class MultipleSelectOptionList extends Component {
    constructor(props) {
        super(props);
    }

    static propTypes = {
        dataSource: PropTypes.arrayOf(PropTypes.object).isRequired,
        id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
        onChange: PropTypes.func
    }

    _renderOptionList() {
        const { dataSource, onChange, id } = this.props;
        if(!dataSource || !dataSource.length){
            return <span>No information</span>
        }

        const optionList = dataSource.map((item, index) => {
            return (
                <Option
                    id={`${id}-optionItem-${index}`}
                    key={`${id}optionItem${index}`}
                    itemData={item}
                    onChange={onChange} />
            )
        });

        return optionList;
    }

    render() {
        return (
            <ul className="multiple-select-options">
                {this._renderOptionList()}
            </ul>
        )
    }
}