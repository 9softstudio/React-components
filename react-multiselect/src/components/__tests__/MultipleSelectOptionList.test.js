import React from 'react'
import renderer from 'react-test-renderer'

import MultipleSelectOptionList from '../MultipleSelectOptionList'

const defaultProps = {
    dataSource: [],
    id: 1
}

describe('MultipleSelectOptionList ', () => {
    afterAll(() => {
        jest.resetAllMocks();
    })

    describe('_renderOptionList', () => {
        it('Option component count should be correct', () => {
            const expectedResult = 2;
            const dataSource = [{
                key: "key1",
                value: "Value 1",
                checked: true
            },
            {
                key: "key2",
                value: "Value 2",
                checked: false
            }
            ];

            const props = Object.assign({}, defaultProps, { dataSource });
            const optionList = new MultipleSelectOptionList(props);

            const result = optionList._renderOptionList();

            expect(result.length).toBe(expectedResult);
        })

        it('props of Option component should be correct', () => {
            const option = {
                key: "key1",
                value: "Value 1",
                checked: true
            };
            const onChangeFunc = jest.fn();

            const expectedResult = {
                id: "1-optionItem-0",
                itemData: option,
                onChange: onChangeFunc
            };

            const dataSource = [option];

            const props = Object.assign({}, defaultProps, { dataSource, onChange: onChangeFunc });
            const optionList = new MultipleSelectOptionList(props);

            const renderedOptionList = optionList._renderOptionList();
            const result = renderedOptionList[0].props;

            expect(result).toEqual(expectedResult);
        })
    })

    it('rendered content should match with snapshot', () => {
        const dataSource = [{
            key: "key1",
            value: "Value 1",
            checked: true
        },
        {
            key: "key2",
            value: "Value 2",
            checked: false
        }
        ];

        const props = Object.assign({}, defaultProps, { dataSource });
        const optionList = renderer.create(<MultipleSelectOptionList {...props} />);

        const result = optionList.toJSON();

        expect(result).toMatchSnapshot();
    })
})