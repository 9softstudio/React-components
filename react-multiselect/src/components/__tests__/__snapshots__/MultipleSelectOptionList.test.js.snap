// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleSelectOptionList  rendered content should match with snapshot 1`] = `
<ul
  className="multiple-select-options"
>
  <li
    className="multiple-select-item"
  >
    <input
      checked={true}
      className="option-checkbox"
      id="1-optionItem-0"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="option-label"
      htmlFor="1-optionItem-0"
    >
      Value 1
    </label>
  </li>
  <li
    className="multiple-select-item"
  >
    <input
      checked={false}
      className="option-checkbox"
      id="1-optionItem-1"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="option-label"
      htmlFor="1-optionItem-1"
    >
      Value 2
    </label>
  </li>
</ul>
`;
